### PR TITLE
feat: Add `--wait-timeout` option for `docker.compose.up`

### PR DESCRIPTION
**Pull request by Taragolis**
This PR add support for wait timeout, https://docs.docker.com/reference/cli/docker/compose/up/#options

```console
--wait-timeout		Maximum duration to wait for the project to be running|healthy
```

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
I seems that the version of docker compose we use in the CI is not recent enough. Maybe we should update it?

**Comment by Taragolis** [comment]
Yeah it added in [`v2.17.0`](https://github.com/docker/compose/releases/tag/v2.17.0), let me check locally that compose tests pass in that version

**Comment by Taragolis** [comment]
I've check it locally with 2.17.0 and 2.17.2 and two tests was failed
```console
================================================================== short test summary info ===================================================================
FAILED tests/python_on_whales/components/test_compose.py::test_project_directory_env_files - python_on_whales.exceptions.DockerException: The docker command executed was `/usr/bin/docker compose --file /home/taragolis/Projects/common/python-on-wh...
FAILED tests/python_on_whales/components/test_compose.py::test_compose_logs_follow - assert 1 >= 5
========================================================== 2 failed, 71 passed in 312.91s (0:05:12) ==========================================================
```

**Comment by Taragolis** [comment]
I could have a look on this tests and try to adopt tests to the newer version of  docker-compose. Any preference to the target version of docker-compose?

**Comment by gabrieldemarmiesse** [comment]
The latest one is fine by me

**Comment by gabrieldemarmiesse** [comment]
@MisterOwlPT  updated the compose version in the CI, let's see if that fixes the tests here

File: python_on_whales/components/compose/cli_wrapper.py [hunk 1]
```diff
@@ -1005,6 +1009,8 @@ def up(
         full_cmd.add_flag("--build", build)
         full_cmd.add_flag("--detach", detach)
         full_cmd.add_flag("--wait", wait)
+        if wait_timeout:
+            full_cmd.add_simple_arg("--wait-timeout", wait_timeout)
```
Comment (gabrieldemarmiesse) [inline python_on_whales/components/compose/cli_wrapper.py:1013]:
```suggestion
        full_cmd.add_simple_arg("--wait-timeout", wait_timeout)
```
`àdd_simple_arg` doesn't do anything if the second argument is None

**Review by gabrieldemarmiesse — CHANGES_REQUESTED** [review]

**Review by gabrieldemarmiesse — APPROVED** [review]
Many thanks for the pull request!

**Comment by gabrieldemarmiesse** [comment]
dang this PR is cursed

**Pull request merged**